packages/dbs-ai/server/methods/getSmartiUiScript.js

```diff
@@ -1,36 +1,58 @@
 /* globals SystemLogger */
 
 let script;
+let timeoutHandle;
 
-Meteor.methods({
-	getSmartiUiScript() {
-		if (script) { //buffering
-			return script;
-		}
 
-		if (!RocketChat.settings.get('DBS_AI_Redlink_URL').trim()) {
-			throw new Meteor.Error('no-smarti-url-configured');
-		}
+function loadSmarti() {
+	if (!RocketChat.settings.get('DBS_AI_Redlink_URL').trim()) {
+		throw new Meteor.Error('no-smarti-url-configured');
+	}
 
-		const DBS_AI_SMARTI_URL =
-			RocketChat.settings.get('DBS_AI_Redlink_URL').endsWith('/') ?
-				RocketChat.settings.get('DBS_AI_Redlink_URL') :
-				`${ RocketChat.settings.get('DBS_AI_Redlink_URL') }/`;
-		SystemLogger.debug('Trying to retrieve Smarti UI from', DBS_AI_SMARTI_URL);
-
-		const response = HTTP.get(`${ DBS_AI_SMARTI_URL }plugin/v1/rocket.chat.js`);
-		if (response.statusCode === 200) {
-			script = response.content;
-
-			if (!script) {
-				SystemLogger.error('Could not extract script from Smarti response');
-				throw new Meteor.Error('no-smarti-ui-script');
-			} else {
-				return script;
-			}
-		} else {
-			SystemLogger.error('Could not reach Smarti service at', DBS_AI_SMARTI_URL);
+	const DBS_AI_SMARTI_URL =
+		RocketChat.settings.get('DBS_AI_Redlink_URL').endsWith('/') ?
+			RocketChat.settings.get('DBS_AI_Redlink_URL') :
+			`${ RocketChat.settings.get('DBS_AI_Redlink_URL') }/`;
+	SystemLogger.debug('Trying to retrieve Smarti UI from', DBS_AI_SMARTI_URL);
+
+	const response = HTTP.get(`${ DBS_AI_SMARTI_URL }plugin/v1/rocket.chat.js`);
+	if (response.statusCode === 200) {
+		script = response.content;
+
+		if (!script) {
+			SystemLogger.error('Could not extract script from Smarti response');
 			throw new Meteor.Error('no-smarti-ui-script');
+		} else {
+			// add pseudo comment in order to make the script appear in the frontend as a file. This makes it de-buggable
+			script = `${ script } //# sourceURL=SmartiWidget.js`;
```

Comment:
This suffix after the script body makes a resource named `SmartiWidget.js` appear in the IDE
Webstorm:
<img width="550" alt="smarti widget webstorm" src="https://user-images.githubusercontent.com/17176678/29308315-b859697e-81a5-11e7-8524-b470a64534a6.png">

Chrome:
<img width="1048" alt="debug smarti" src="https://user-images.githubusercontent.com/17176678/29308227-45555c08-81a5-11e7-9599-90d8341f691e.png">